src/server.js

```diff
@@ -1,52 +1,53 @@
 #!/usr/bin/env node
 'use strict';
 
+import os from 'os';
 process.env.UV_THREADPOOL_SIZE =
-    Math.ceil(Math.max(4, require('os').cpus().length * 1.5));
-
-const fs = require('fs');
-const path = require('path');
-
-const chokidar = require('chokidar');
-const clone = require('clone');
-const cors = require('cors');
-const enableShutdown = require('http-shutdown');
-const express = require('express');
-const handlebars = require('handlebars');
-const mercator = new (require('@mapbox/sphericalmercator'))();
-const morgan = require('morgan');
-
-const packageJson = require('../package');
-const serve_font = require('./serve_font');
-const serve_style = require('./serve_style');
-const serve_data = require('./serve_data');
-const utils = require('./utils');
-
-let serve_rendered = null;
+    Math.ceil(Math.max(4, os.cpus().length * 1.5));
+
+import fs from 'node:fs';
+import path from 'path';
+
+import chokidar from 'chokidar';
+import clone from 'clone';
+import cors from 'cors';
+import enableShutdown from 'http-shutdown';
+import express from 'express';
+import handlebars from 'handlebars';
+import SphericalMercator from '@mapbox/sphericalmercator';
+const mercator = new SphericalMercator();
+import morgan from 'morgan';
+import {serve_data} from './serve_data.js';
+import {serve_style} from './serve_style.js';
+import {serve_font} from './serve_font.js';
+import {getTileUrls, getPublicUrl} from './utils.js';
+
+import {fileURLToPath} from 'url';
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const packageJson = JSON.parse(fs.readFileSync(__dirname + '/../package.json', 'utf8'));
+
 const isLight = packageJson.name.slice(-6) === '-light';
-if (!isLight) {
-  // do not require `serve_rendered` in the light package
-  serve_rendered = require('./serve_rendered');
-}
+const serve_rendered = (await import(`${!isLight ? `./serve_rendered.js` : `./serve_light.js`}`)).serve_rendered;
```

Comment:
I think this is actually the optimal solution for now.

The other idea I floated was to have two entrypoints (`src/main.js` and `src/light.js` or something) and `main.js` would import `server_rendered.js` while `light.js` would include the stub, then they each would import `src/server.js` and pass `serve_rendered` method to it. But in its current state it'd be a lot of duplication and a fair amount of passing `serve_rendered` around, or a global variable or something. Neither is better than what we have here, IMO.